cli/internal/analytics/client.go

```diff
@@ -0,0 +1,229 @@
+		environment: getEnvironment(),
+	}
+}
+
+func Identify(ctx context.Context, invocationUUID uuid.UUID) {
+	if client == nil {
+		return
+	}
+
+	details := getEventDetails(ctx)
+	if details == nil {
+		_ = client.Enqueue(rudderstack.Identify{
+			AnonymousId: invocationUUID.String(),
+			Traits: rudderstack.Traits{
+				"environment": getEnvironment(),
+			},
+		})
+		return
+	}
+
+	err := client.Enqueue(rudderstack.Identify{
+		AnonymousId: invocationUUID.String(),
+		UserId:      details.user.ID.String(),
+	})
+	if err != nil {
+		return
+	}
+
+	_ = client.Enqueue(rudderstack.Group{
+		UserId:  details.user.ID.String(),
+		GroupId: details.currentTeam,
+	})
+}
+
+func TrackCommandStart(ctx context.Context, commandName string, invocationUUID uuid.UUID) {
+	if client == nil {
+		return
+	}
+
+	_ = client.Enqueue(rudderstack.Track{
+		AnonymousId: invocationUUID.String(),
+		Event:       "command_" + strings.ToLower(commandName) + "_start",
+		Properties: rudderstack.Properties{
+			"invocation_uuid": invocationUUID,
+		},
+	})
+}
+
+func TrackCommandEnd(ctx context.Context, commandName string, invocationUUID uuid.UUID, err error) {
+	if client == nil {
+		return
+	}
+
+	status := "success"
+	if err != nil {
+		status = "error"
+	}
+
+	_ = client.Enqueue(rudderstack.Track{
+		AnonymousId: invocationUUID.String(),
+		Event:       "command_" + strings.ToLower(commandName) + "_end",
+		Properties: rudderstack.Properties{
+			"invocation_uuid": invocationUUID,
+			"status":          status,
+		},
+	})
+}
+
+func TrackLoginSuccess(ctx context.Context, invocationUUID uuid.UUID) {
+	if client == nil {
+		return
+	}
+
+	details := getEventDetails(ctx)
+	if details == nil {
+		return
+	}
+
+	_ = client.Enqueue(rudderstack.Track{
+		AnonymousId: invocationUUID.String(),
+		UserId:      details.user.ID.String(),
+		Event:       "login_success",
+		Properties: rudderstack.Properties{
+			"invocation_uuid": invocationUUID,
+			"team":            details.currentTeam,
+			"environment":     details.environment,
+		},
+	})
+}
+
+type SyncStartedEvent struct {
+	Source       specs.Source
+	Destinations []specs.Destination
+}
+
+func getSyncCommonProps(invocationUUID uuid.UUID, event SyncStartedEvent, details *eventDetails) rudderstack.Properties {
+	destinationPaths := make([]string, len(event.Destinations))
+	for i, d := range event.Destinations {
+		destinationPaths[i] = d.Path
+	}
+
+	props := rudderstack.NewProperties().
+		// we are using the same invocation_uuid for sync_run_id
+		// invocation_uuid to be consistent with the rest of the events
+		// sync_run_id to match with cloud events
+		Set("invocation_uuid", invocationUUID).
+		Set("sync_run_id", invocationUUID).
+		Set("team", details.currentTeam).
+		Set("environment", details.environment).
+		Set("sync_name", event.Source.Name).
+		Set("source_path", event.Source.Path).
+		Set("destination_paths", destinationPaths).
+		Set("user_id", details.user.ID).
+		Set("user_email", details.user.Email)
```

Comment:
I missed this in my first review, but should we have sync group ID here (or somewhere) as well?

Reply:
Do you mean https://docs.cloudquery.io/docs/reference/destination-spec#sync_group_id-preview? It's per destination, so I think if we add that maybe we should add an additional property with `source_spec` and `destinations_specs` with all the options we'd like to track.

Reply:
Yeah we can leave it out for now, but I thought it would be nice to know when parallel syncs are related to one another.